lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -193,8 +194,40 @@ func (r *forwardInterceptor) resolveFromClient(
 	switch in.Action {
 	case ResolveHoldForwardAction_RESUME:
 		return interceptedForward.Resume()
+
 	case ResolveHoldForwardAction_FAIL:
-		return interceptedForward.Fail()
+		// Fail with an encrypted reason.
+		if in.FailureMessage != nil {
+			if in.FailureCode != 0 {
+				return errors.New("failure message and " +
+					"failure code are mutually exclusive")
```

Comment:
non blocking: add `INVALID_ARGUMENT` status? Although lnd isn't consistent on error codes, nice to add them when we have new errors imo.

Reply:
Added here and in a few other places.